File: backend/core-service/src/Fix.Application/UseCases/Mandates/Dtos/MandateDto.cs
using Fix.Domain.Common;
using Fix.Domain.AggregateRoots.Mandates;

namespace Fix.Application.Mandates.Dtos;

public sealed record MandateDto(
    Guid Id,
    Guid PolicyId,
    string PolicyCode,
    string PolicyVersion,
    Guid AxisId,
    string AxisCode,
    string AxisTitle,
    MandateType Type,
    string Title,
    string? Criteria,
    Commodity? Commodity,
    string? Tenor,
    decimal? Quantity,
    MeasurementUnit? QuantityUnit,
    // Consumido por boletas aprovadas (lotes ou US$).
    decimal Consumed,
    // Saldo disponível; nulo quando o mandato não tem teto de volume.
    decimal? Balance,
    PriceCriteriaDto Price,
    DateOnly? WindowStart,
    DateOnly? WindowEnd,
    ComplianceDto Compliance,
    MandateStatus Status,
    Guid IssuedBy,
    Guid? DecidedBy,
    DateTimeOffset? DecidedAt,
    string? DecisionNote);

File: frontend/bff/src/application/mandates/dtos/mandate.dto.ts
import type { ComplianceDto } from './compliance.dto.js';
import type { MandateStatus } from '../../../cross-cutting/enums/mandate-status.js';
import type { MandateTermsDto } from './mandate-terms.dto.js';
import type { MandateType } from '../../../cross-cutting/enums/mandate-type.js';

export interface MandateDto {
  id: string;
  /** Código legível da organização (MD-01). */
  code: string;
  policyId: string;
  policyCode: string;
  policyVersion: string;
  axisId: string;
  axisCode: string;
  axisTitle: string;
  type: MandateType;
  terms: MandateTermsDto;
  /** Consumido por boletas aprovadas (lotes ou US$). */
  consumed: number;
  /** Saldo; null = sem teto de volume. */
  balance: number | null;
  compliance: ComplianceDto;
  status: MandateStatus;
  issuedBy: string;
  decidedBy: string | null;
  decidedAt: string | null;
  decisionNote: string | null;
}
